### PR TITLE
changes for logging round data to csv

### PR DESCRIPTION
**Pull request by Noanan**
Link to colab: https://colab.research.google.com/drive/13tvBXKd1afLvrUNcDyXaXffvD7vl3AP2?usp=sharing

To get viz: 

- run the code
- get the generated csv file (timestamped HH-MM)
- upload it to colab
- change the input file path (right click file in colab file explorer in the left pane and use "copy path")

### CODE REVIEW

**Comment by antonypap** [comment]
Why not have it as a `.py` file (instead of the notebook) and save them directly to a directory, since we already have the `.csv`s here anyway?

**Comment by Noanan** [comment]
> Why not have it as a `.py` file (instead of the notebook) and save them directly to a directory, since we already have the `.csv`s here anyway?

though against it bc it would mean making a python env with all the modules which is prob more time consuming and fustrating to fix then just having to upload it.

down to make a .py file and change it if you guys think its more convenient

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ logs/*
 .idea
 .DS_Store
 coverage.txt
+.csv
```
Comment (MattSScott) [inline .gitignore]: *.csv

File: pkg/infra/main.go [hunk 2]
```diff
@@ -54,6 +57,22 @@ func startGameLoop() {
...
+		log.Fatalf("failed creating file: %s", err)
+	}
+
+	w := csv.NewWriter(csvFile)
+	// title row
+	firstRow := []string{"level", "total agents alive", "average health", "average stamina", "average attack", "average defense", "average personality", "average sanctioned", "count selfless", "count selfish", "count collective"}
+	if err := w.Write(firstRow); err != nil {
+		log.Fatalln("error writing record to file", err)
+	}
+
```
Comment (MattSScott) [inline pkg/infra/main.go]: Plz put in separate function for readability

**Comment by MattSScott** [comment]
It's probably best leaving all of the csv stuff inside of agent 3, since it shouldn't really be a global function. At the same time, idk who else is going to use this repo, so it's probably not a huge deal

**Review by MattSScott — APPROVED** [review]

**Comment by antonypap** [comment]
> > Why not have it as a `.py` file (instead of the notebook) and save them directly to a directory, since we already have the `.csv`s here anyway?
> 
> though against it bc it would mean making a python env with all the modules which is prob more time consuming and fustrating to fix then just having to upload it.
> 
> down to make a .py file and change it if you guys think its more convenient

Yeah I see what you mean, still though I think it's a better solution (btw it's only 3 libraries that I think most of us should have on their computer already, so it should be fine for now). 
Maybe just export the `.ipynb` to a `.py` and add saving to a dir here as well?

Also maybe move the `csv`s into a separate dir, just to keep the working dir clean?

**Comment by Noanan** [comment]
> > > Why not have it as a `.py` file (instead of the notebook) and save them directly to a directory, since we already have the `.csv`s here anyway?
> > 
> > 
> > though against it bc it would mean making a python env with all the modules which is prob more time consuming and fustrating to fix then just having to upload it.
> > down to make a .py file and change it if you guys think its more convenient
> 
> Yeah I see what you mean, still though I think it's a better solution (btw it's only 3 libraries that I think most of us should have on their computer already, so it should be fine for now). Maybe just export the `.ipynb` to a `.py` and add saving to a dir here as well?
> 
> Also maybe move the `csv`s into a separate dir, just to keep the working dir clean?

done

python code should be good, ported directly from the colab. Just coudn't test it bc wsl dosen't have a GUI for the plots :)

**Pull request merged**